webhook/resources/node/validator.go

```diff
@@ -71,6 +71,11 @@ func (n *nodeValidator) Update(request *admission.Request, oldObj runtime.Object
 			oldNode.Name), "")
 	}
 
+	// Ensure the node controller already syncs the disk spec and status.
+	if len(oldNode.Spec.Disks) != len(oldNode.Status.DiskStatus) {
```

Comment:
It's better to check if the entries in `Status.DiskStatus` exists in `Spec.Disks`.
This check cannot prevent the following case:
```
Spec.Disk: Disk A, Disk B
Status.DiskStatus: Disk A, Disk C
```

Reply:
Thank you. I was not aware of the case....

Reply:
I thought it should be finally consistent? Should we need to worry this case?

Reply:
Yeah. This is for the corner case when node controller failed to reconcile node before `alignDiskSpecAndStatus`.

Reply:
I see, yeah we need to check the content as well not just size.

Reply:
Addressed in https://github.com/longhorn/longhorn-manager/pull/1311.